backend/safe_condition_parser.py

```
@@ -389,12 +389,9 @@ def safe_evaluate_condition(
     """
     try:
         return _parser.evaluate(expression, context)
-    except ValueError:
-        # Validation and safety errors from the parser should surface to callers/tests
-        raise
     except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)